File: spring-only/src/main/java/com/akvone/lifecycle/MainEventListener.java
package com.akvone.lifecycle;

import org.springframework.context.event.ContextClosedEvent;
import org.springframework.context.event.EventListener;
import org.springframework.stereotype.Component;

@Component
public class MainEventListener {

  @EventListener(ContextClosedEvent.class)
  public void onContextClosedEvent(ContextClosedEvent event){
    System.out.println(event);
  }
}

File: spring-boot/src/main/kotlin/com/akvone/lifecycle/MainEventListener.kt
package com.akvone.lifecycle

import org.slf4j.LoggerFactory
import org.springframework.context.event.ContextClosedEvent
import org.springframework.context.event.EventListener
import org.springframework.stereotype.Component

@Component
class MainEventListener {
    private val log = LoggerFactory.getLogger(MainEventListener::class.java)

    @EventListener(ContextClosedEvent::class)
    fun onContextClosedEvent(event: ContextClosedEvent?) {
        log.info(event.toString())
    }
}